module/user-extension/helper.js

```diff
@@ -19,670 +18,825 @@ const programsHelper = require(MODULES_BASE_PATH + "/programs/helper");
-                            "isDeleted": false,
-                            "devices": [deviceData],
-                            "createdBy": "SYSTEM",
-                            "updatedBy": "SYSTEM"
-                        }
-                    );
-
-                    if (createUserExtension) {
-                        response["success"] = true;
-                        response["message"] =
-                            `Successfully created user ${userDetails.userId} in userExtension`;
-                    } else {
-                        throw `Could not create user ${userDetails.userId} in userExtension`;
-                    }
-
-                }
-
-                return resolve(response);
-
-            } catch (error) {
-                return reject(error);
+          } else {
+            deviceNotFound = true;
+          }
+
+          if (deviceNotFound) {
+            let updatedData = await database.models.userExtension
+              .findOneAndUpdate(
+                {
+                  userId: userDetails.userId,
+                },
+                { $addToSet: { devices: deviceData } }
+              )
+              .lean();
+
+            if (updatedData) {
+              response["success"] = true;
+              response[
+                "message"
+              ] = `Added Successfully device id ${deviceData.deviceId} for user ${userDetails.email}`;
+            } else {
+              throw `Could not add device id ${deviceData.deviceId} for user ${userDetails.email}`;
             }
-        })
-    }
-
-    /**
-     * Update device status in userExtension document.
-     * @method
-     * @name updateDeviceStatus
-     * @name updateDeviceStatus  
-     * @param {Object} deviceData - device data.
-     * @param {String} deviceData.title - title of device.
-     * @param {Object[]} deviceArray - device array.      * 
-     * @param {String} userId - Logged in user id.      
-     * @returns {Promise} returns a promise.
-    */
-
-    static updateDeviceStatus(deviceId, deviceArray, userId) {
-
-        return new Promise(async (resolve, reject) => {
-
-            try {
-                deviceArray.forEach(async devices => {
-
-                    delete devices['message'];
-                    delete devices['title'];
-
-                    if (devices.deviceId == deviceId) {
-                        devices.status = "inactive";
-                        devices.deactivatedAt = new Date();
-                    }
-                });
-
-                let updateDevice = await database.models.userExtension.findOneAndUpdate(
-                    { userId: userId },
-                    { $set: { "devices": deviceArray } }
-                ).lean();
-
-                if (!updateDevice) {
-                    throw "Could not update device.";
-                }
+          }
+        } else {
+          let createUserExtension = await database.models.userExtension.create({
```

Comment:
@praveenKDass write a function to create the userExtension instead of directly using model